### PR TITLE
Wait generate token when initialising jmap client

### PR DESCRIPTION
**Pull request by alagane**
Related to https://github.com/linagora/customers/issues/55 and #482

Needs https://github.com/linagora/esn-frontend-common-libs/pull/225 to be merged first

### CODE REVIEW

**Comment by oubchid** [comment]
tests failed

**Comment by oubchid** [comment]
Should merge this after
https://github.com/linagora/esn-frontend-inbox/issues/482

**Review by rezk2ll — APPROVED** [review]

**Comment by rezk2ll** [comment]
> Should merge this after
> #482

why? if we merge this PR the #482 will be resolved

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js [hunk 1]
```diff
@@ -41,6 +44,7 @@ describe('The jmapClientProvider service', function() {
     var error = new Error('error message');
 
     tokenAPIMock.getWebToken = () => $q.reject(error);
```
Comment (tuanlc) [inline src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js:46]: then this mock is useless

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js [hunk 2]
```diff
@@ -55,6 +59,7 @@ describe('The jmapClientProvider service', function() {
     config['linagora.esn.unifiedinbox.api'] = 'expected jmap api';
     config['linagora.esn.unifiedinbox.downloadUrl'] = 'expected jmap downloadUrl';
     tokenAPIMock.getWebToken = () => $q.when({ data: 'jwt' });
```
Comment (tuanlc) [inline src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js:61]: same here, this mock is useless

File: src/linagora.esn.unifiedinbox/app/components/inbox-configuration/filter-definition/configuration-filter-definition.controller.spec.js [hunk 3]
```diff
@@ -27,7 +27,14 @@ describe('The inboxConfigurationFilterDefinitionController', function() {
         })
       };
 
+      esnAuth = {
+        signInCompletePromise: {
+          then: callback => callback()
```
Comment (tuanlc) [inline src/linagora.esn.unifiedinbox/app/components/inbox-configuration/filter-definition/configuration-filter-definition.controller.spec.js:32]: Or you can use:
```
signInCompletePromise: $q.when()
```

File: src/linagora.esn.unifiedinbox/app/components/mailbox-shared-settings/mailbox-shared-settings.controller.spec.js [hunk 4]
```diff
@@ -43,12 +44,19 @@ describe('The InboxMailboxSharedSettingsController controller', function() {
         return user;
       })
     };
+
+    esnAuth = {
+      signInCompletePromise: {
+        then: callback => callback()
+      }
```
Comment (tuanlc) [inline src/linagora.esn.unifiedinbox/app/components/mailbox-shared-settings/mailbox-shared-settings.controller.spec.js:51]: same here, you just need to return a resolved promise

```
signInCompletePromise: $q.when()
```

Comment (alagane) [reply src/linagora.esn.unifiedinbox/app/components/mailbox-shared-settings/mailbox-shared-settings.controller.spec.js:51]: I tried, but it did not work, I can still try again to make sure.

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js (hunk 1 above)
Comment (alagane) [reply src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js]: I removed it

File: src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js (hunk 2 above)
Comment (alagane) [reply src/esn.inbox.libs/app/services/jmap-client-provider/jmap-client-provider.spec.js]: I removed it too

File: src/linagora.esn.unifiedinbox/app/components/mailbox-shared-settings/mailbox-shared-settings.controller.spec.js (hunk 4 above)
Comment (alagane) [reply src/linagora.esn.unifiedinbox/app/components/mailbox-shared-settings/mailbox-shared-settings.controller.spec.js]: Ok, it works, so I replaced it.

File: src/linagora.esn.unifiedinbox/app/components/inbox-configuration/filter-definition/configuration-filter-definition.controller.spec.js (hunk 3 above)
Comment (alagane) [reply src/linagora.esn.unifiedinbox/app/components/inbox-configuration/filter-definition/configuration-filter-definition.controller.spec.js]: I replaced it

**Review by tuanlc — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Pull request merged**